### ISSUE TITLE
intersectBezier2Bezier3 cannot find all intersections for some case

### ISSUE THREAD

**Issue by hnakamur** [thelonious/kld-intersections#2]
First of all, thanks for a great software!

I found that intersectBezier2Bezier3 cannot find all intersections for some case.
I added two test cases, one passes and another fails.
https://github.com/hnakamur/kld-intersections/commit/5070de0a1005a63db0ebe1c3bf33bbd24305ef77

You can try the demo at http://bl.ocks.org/hnakamur/73af30f05fced219a33f
Please check intersections which are printed with console.log().
The demo source is at https://github.com/hnakamur/d3.js-drag-bezier-curves-example/tree/use-kld-intersections

**Comment by Quazistax** [comment]
Try increasing bisection accuracy parameter in Polynomial.js

`Polynomial.ACCURACY = 15`

https://github.com/Quazistax/kld-polynomial/commit/e13cc9aba4f487812e243551f1b534213199c74a
It worked for that case in your demo. Let me know if you stumble upon some other problem.

**Comment by hnakamur** [comment]
Hi, thanks for your modification.
I tried my demo with `Polynomial.ACCURACY = 15`.
However I had a missing intersection near p1-p2 on the cubic bezier curve.
![missing_intersection_on_p1-p2](https://cloud.githubusercontent.com/assets/19299/5694843/e3aa88a4-99bf-11e4-8701-650c97faec07.png)

When I moved p4 on the cubic bezier curve, the left intersection was found or not found.
p4: 143/200 intersection not found
p4: 144/200 intersection not found
p4: 145/200 intersection not found
p4: 146/200 intersection found
p4: 147/200 intersection not found

**Comment by Quazistax** [comment]
Your demo at http://bl.ocks.org/hnakamur/raw/73af30f05fced219a33f/ still uses `ACCURACY = 6`. With that setting, for some combination of parameters (points), calculation error gets too big and result does not register as intersection.

 I agree that even with `ACCURACY = 15` there could be some rare (although more rare) cases where same problem occurs. If you encounter one, let me know. That kind of uncertainty could be eliminated by using different intersection algorithm, for example by approximating curves using lines, but I didn't implement that kind of solution yet.

You don need to set `ACCURACY` to more than 15. 'Double' floating point number cannot hold more digits. 

Useful post: https://stackoverflow.com/questions/16052689/how-to-find-intersection-points-between-two-cubic-bezier-curve

**Comment by hnakamur** [comment]
Thanks for pointing me out for not updating my demo. 
Now I updated my demo at http://bl.ocks.org/hnakamur/raw/73af30f05fced219a33f/ with Polynomial.ACCURACY = 15

And yes, now my demo shows all intersections!

Actually I had implemented an example by approximating curves using lines.
demo: http://bl.ocks.org/hnakamur/7dfb57d5f299255cf59a
source: https://github.com/hnakamur/d3.js-drag-bezier-curves-example/tree/fad37c282c9e8457366544abf93702ac4cb38523

I used the algorithm described in "7.6 Interval subdivision", "Figure 7.3: Interval preprocess and subdivision" of http://tom.cs.byu.edu/~557/text/cagd.pdf

Now I close this pull request.
Thanks!
